File: ide/src/main/java/org/dau/ide/schema/SchemaActions.java
package org.dau.ide.schema;

import org.dau.ide.action.FxAction;
import org.dau.ide.action.ToolbarAction;
import org.dau.ide.project.ProjectConf;
import org.springframework.stereotype.Component;

@Component
public final class SchemaActions {

  @SchemaBean
  @ToolbarAction
  public FxAction saveAction(SchemaConf conf, ProjectConf projectConf) {
    return new FxAction("icons/save.png", "Save")
      .on(() -> {
        var file = projectConf.directory.resolve(conf.schema.toFileName());
        conf.schema.save(file);
      });
  }
}

File: ide/src/main/kotlin/org/dau/ide/schema/SchemaActions.kt
package org.dau.ide.schema

import org.dau.ui.action.FxAction
import org.dau.ui.action.ToolbarAction
import org.dau.ide.project.ProjectConf
import org.springframework.stereotype.Component

@Component
class SchemaActions {

  @SchemaBean
  @ToolbarAction
  fun saveAction(conf: SchemaConf, projectConf: ProjectConf): FxAction {
    return FxAction("icons/save.png", "Save")
      .on {
        val file = projectConf.directory.resolve(conf.schema.toFileName())
        conf.schema.save(file)
      }
  }
}
